api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationCustom.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.api.kafka.model.GenericSecretSource;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Configures a listener to use mutual TLS authentication.
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaListenerAuthenticationCustom extends KafkaListenerAuthentication {
+
+    public static final String FORBIDDEN_PREFIXES = "ssl.keystore.location, ssl.keystore.password, ssl.keystore.type";
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String TYPE_CUSTOM = "custom";
+
+    private Map<String, Object> listenerConfig;
+    private boolean sasl;
+    private List<GenericSecretSource> secrets;
+
+    @Description("Must be `" + TYPE_CUSTOM + "`")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Override
+    public String getType() {
+        return TYPE_CUSTOM;
+    }
+
+    @Description("Enable or disable SASL on this listener.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public boolean isSasl() {
+        return sasl;
+    }
+
+    public void setSasl(boolean enabled) {
+        this.sasl = enabled;
+    }
+
+    @Description("Configuration to be used for a specific listener. All values are prefixed with lister.name.<lister-name>.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public Map<String, Object> getListenerConfig() {
+        return listenerConfig;
+    }
+
+    public void setListenerConfig(Map<String, Object> config) {
+        this.listenerConfig = config;
+    }
+
+    @Description("Secrets to be mounted to /mnt/strimzi/custom-auth-secrets/<listener-name>/<secret-key>")
```

Comment:
I think this is another wrong path. FRom the code it seems like `/opt/kafka/custom-auth-secrets/custom-<listener-name>/<secret-name>` or something like that. I think you have the right path in `io.strimzi.api.kafka.model.listener.KafkaListenerAuthenticationCustom.adoc`, just need to fix it here as well.

Reply:
Yeah re-thinking this, should we mount it to `/opt/kafka/custom-auth-secrets/custom-listener-<listener-name>/<secret-name>` instead? (Only difference, is the folder name `custom-<listener-name>` vs `custom-listener-<listener-name>`

Reply:
Also, what do we think is more clear, `custom-<listener-name>-<port>` or `custom-<listener-name>`? Technically, it's the former, but the <listener-name> maybe assumed to be listener name + port?

Reply:
I'm flexible TBH. I think the `/opt/kafka/custom-auth-secrets` is important since that defines what secrets these are. Having both the listener name and port in the name helps to orient your self better from my opinion. I think other paths use the `listener-name-port` format as well.